packages/build-tools/src/common/easBuildInternal.ts

```diff
@@ -0,0 +1,127 @@
+import assert from 'assert';
+
+import { Env, Job, Metadata, sanitizeJob, sanitizeMetadata } from '@expo/eas-build-job';
+import { PipeMode } from '@expo/logger';
+import spawn from '@expo/turtle-spawn';
+import Joi from 'joi';
+
+import { BuildContext } from '../context';
+
+const EasBuildInternalResultSchema = Joi.object<{ job: object; metadata: object }>({
+  job: Joi.object().unknown(),
+  metadata: Joi.object().unknown(),
+});
+
+export async function runEasBuildInternalAsync<TJob extends Job>(
+  ctx: BuildContext<TJob>
+): Promise<void> {
+  const { cmd, args } = resolveEasCommandPrefix();
+  const buildProfile = ctx.job.buildProfile;
+  assert(buildProfile, 'build profile is missing in a build from git.');
+  const result = await spawn(
+    cmd,
+    [...args, 'build:internal', '--platform', ctx.job.platform, '--profile', buildProfile],
+    {
+      cwd: ctx.reactNativeProjectDirectory,
+      env: { ...ctx.env, EXPO_TOKEN: ctx.job.secrets.robotAccessToken },
+      logger: ctx.logger,
+      mode: PipeMode.STDERR_ONLY_AS_STDOUT,
+    }
+  );
+  try {
+    const stdout = result.stdout.toString();
+    const parsed = JSON.parse(stdout);
+    const { job, metadata } = validateEasBuildInternalResult(ctx, parsed);
+    ctx.updateJobInformation(job, metadata);
+  } catch (err) {
+    throw err;
+  }
+}
+
+export async function configureEnvFromBuildProfileAsync<TJob extends Job>(
+  ctx: BuildContext<TJob>
+): Promise<void> {
+  const { cmd, args } = resolveEasCommandPrefix();
+  const buildProfileName = ctx.job.buildProfile;
+  assert(buildProfileName, 'build profile is missing in a build from git.');
+  let spawnResult;
+  try {
+    spawnResult = await spawn(
+      cmd,
+      [
+        ...args,
+        'config',
+        '--platform',
+        ctx.job.platform,
+        '--profile',
+        buildProfileName,
+        '--non-interactive',
+        '--json',
+        '--eas-json-only',
+      ],
+      {
+        cwd: ctx.reactNativeProjectDirectory,
+        env: ctx.env,
+      }
+    );
+  } catch (err: any) {
+    ctx.logger.error(`Failed to read build profile ${buildProfileName} from eas.json.`);
+    ctx.logger.error(err.stderr?.toString());
+    throw Error(`Failed to read build profile ${buildProfileName} from eas.json.`);
+  }
+  try {
+    assert(spawnResult);
+    const stdout = spawnResult.stdout.toString();
+    const parsed = JSON.parse(stdout);
+    const env = validateEnvs(parsed);
+    ctx.updateEnv(env);
+  } catch (err) {
+    throw err;
+  }
+}
+
+function resolveEasCommandPrefix(): { cmd: string; args: string[] } {
+  if (process.env.ENVIRONMENT === 'development') {
+    return { cmd: process.env.EAS_BUILD_INTERNAL_EXECUTABLE ?? `eas`, args: [] };
+  } else if (process.env.ENVIRONMENT === 'staging') {
+    return { cmd: 'npx', args: ['--yes', 'eas-cli@latest-eas-build-staging'] };
+  } else {
+    return { cmd: 'npx', args: ['--yes', 'eas-cli@latest-eas-build'] };
```

Comment:
What are those tags? Add a comment + update the PR description.

Reply:
Updated PR description, but I'm not sure what you would want me to add in a comment. The meaning of those tags is quite clear.

Reply:
I strongly believe those should be separate tags:
- `latest-eas-build-staging` will be always set to the latest automatically, but I think it is valuable to have a way to easily test different versions on staging, without deploying a modified workers.
- `latest-eas-build` is useful in case we need to rollback, we don't need to release new cli. I would prefer that this step would be manual, but we can discuss automatically setting this tag on release. Or another alternative, maybe we could update this tag every time a worker is deployed to production.

Reply:
I agree with Wojtek - having two separate tags seems a practical and safer option to me (although it's more complicated and harder to maintain). By choosing this approach we can have more control over what's going on in production.